### PR TITLE
feat(rollbar): add error boundaries on app and main view

### PR DESCRIPTION
**Pull request by ArslanSaleem**
## Release Notes

- **New Features**
	- Introduced error handling and monitoring capabilities using Rollbar.
	- Added `ErrorBoundary` and custom fallback components (`GlobalErrorFallback`, `ViewErrorFallback`) for improved error management.
	- New environment variable for Rollbar access token added.
	- Integrated new dependencies for Rollbar support.

- **Bug Fixes**
	- Enhanced error handling to catch and display errors in child components effectively.

- **Documentation**
	- Updated example environment file to include new Rollbar access token variable.

### CODE REVIEW

File: frontend/src/components/ErrorFallback.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    router.push = (...args) => {
+      handleRouteChange(); // Call the reset function
+      return originalPush(...args);
+    };
+
+    // Cleanup function to restore original router push method
+    return () => {
+      router.push = originalPush;
+    };
+  }, [resetError, router]);
```
[the remark this answers is not in the document]
Comment (ArslanSaleem) [reply frontend/src/components/ErrorFallback.tsx:70]: Unhandled Runtime Error
Error: NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted

gives this error

File: frontend/src/app/layout.tsx [hunk 2]
```diff
@@ -4,9 +4,16 @@ import React from "react";
...
 import "@/app/style/globals.css";
+import { Provider, ErrorBoundary } from "@rollbar/react";
+import { GlobalErrorFallback } from "@/components/ErrorFallback";
 
 const inter = Inter({ subsets: ["latin"] });
 
+const rollbarConfig = {
+  accessToken: process.env.NEXT_PUBLIC_ROLLBAR_ACCESS_TOKEN,
+  environment: process.env.NODE_ENV ?? "production",
+};
```
[the remark this answers is not in the document]
Comment (ArslanSaleem) [reply frontend/src/app/layout.tsx:15]: This is intentional to expose

File: frontend/src/components/ErrorFallback.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArslanSaleem) [reply frontend/src/components/ErrorFallback.tsx:40]: The UI get stuck on trying this

[the remark this answers is not in the document]
Comment (gventuri) [reply frontend/src/components/ErrorFallback.tsx:40]: @ArslanSaleem let's fix it somehow, too risky merging this. The code above seems good

File: frontend/src/app/layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gventuri) [reply frontend/src/app/layout.tsx:15]: @ArslanSaleem let's just make sure we are exposing the frontend api key and not the backend api key (they are different in rollbar)

**Pull request merged**